src/pykeen/models/nbase.py

```diff
@@ -429,92 +342,123 @@ def append_weight_regularizer(
 
     def forward(
         self,
-        h_indices: Optional[torch.LongTensor],
-        r_indices: Optional[torch.LongTensor],
-        t_indices: Optional[torch.LongTensor],
+        h_indices: torch.LongTensor,
+        r_indices: torch.LongTensor,
+        t_indices: torch.LongTensor,
         slice_size: Optional[int] = None,
-        slice_dim: Optional[str] = None,
+        slice_dim: int = 0,
     ) -> torch.FloatTensor:
         """Forward pass.
 
-        This method takes head, relation and tail indices and calculates the corresponding score.
-
-        All indices which are not None, have to be either 1-element or have the same shape, which is the batch size.
+        This method takes head, relation and tail indices and calculates the corresponding scores.
+        It supports broadcasting.
 
         :param h_indices:
-            The head indices. None indicates to use all.
+            The head indices.
         :param r_indices:
-            The relation indices. None indicates to use all.
+            The relation indices.
         :param t_indices:
-            The tail indices. None indicates to use all.
+            The tail indices.
+
+        :return:
+            The scores
+        """
+        h, r, t = self._get_representations(h=h_indices, r=r_indices, t=t_indices)
+        return self.interaction.score(h=h, r=r, t=t, slice_size=slice_size, slice_dim=slice_dim)
```

Comment:
why no repeat if necessary here?

Reply:
it is, but the logic is complex. I'll just raise an error for now if repetition would become necessary